.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,50 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches-ignore:
+      - 'dependabot/**'
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Why is this needed?

Reply:
To prevent having multiple CodeQL runs on the same branch. Only the last run is relevant, so we cancel the other ones.